tool-map.nix

```diff
@@ -1,5 +1,13 @@
 # see https://haskell-language-server.readthedocs.io/en/latest/support/ghc-version-support.html
 # so we assume "latest" for all hls. 
 # for hlint however, we need hlint-3.3 for ghc-8.10.7.
-let fixed-versions = { "hlint" = { "ghc8107" = "3.3"; }; }; in
-compiler-nix-name: tool: fixed-versions.${tool}.${compiler-nix-name} or "latest"
\ No newline at end of file
+let fixed-versions = { "hlint" = { "ghc8107" = { version = "3.3"; evalSystem = "x86_64-linux"; }; }; }; in
+compiler-nix-name: tool:
+  if tool == "haskell-language-server"
+    then {pkgs, ...}: rec {
+      src = pkgs.haskell-nix.sources."hls-1.10";
+      cabalProject = __readFile (src + "/cabal.project");
+      sha256map."https://github.com/pepeiborra/ekg-json"."7a0af7a8fd38045fd15fb13445bdcc7085325460" = "sha256-fVwKxGgM0S4Kv/4egVAAiAjV7QB5PBqMVMCfsv7otIQ=";
+      evalSystem = "x86_64-linux";
+    }
```

Comment:
We need a comment here, as to why we do this. What's the logic/reason behind using _this_ approach? We should document that.

And it should probably be it's own PR as well :-/